Plugins/Rhetos.LightDMS/Upload/UploadHelper.cs

```diff
@@ -36,75 +41,48 @@ public class UploadHelper
     {
         private readonly ILogger _performanceLogger;
         private readonly ConnectionString _connectionString;
+        private readonly LightDmsOptions _lightDmsOptions;
+        private readonly S3Options _s3Options;
 
-        public UploadHelper(ILogProvider logProvider, ConnectionString connectionString)
+        public UploadHelper(ILogProvider logProvider, ConnectionString connectionString, LightDmsOptions lightDmsOptions, S3Options s3Options)
         {
             _performanceLogger = logProvider.GetLogger("Performance.LightDMS");
             _connectionString = connectionString;
+            _lightDmsOptions = lightDmsOptions;
+            _s3Options = s3Options;
         }
 
         public async Task<FileUploadResult> UploadStream(Stream inputStream)
         {
             var id = Guid.NewGuid();
             var sw = Stopwatch.StartNew();
-            int bufferSize = 100 * 1024; // 100 kB buffer
-            byte[] buffer = new byte[bufferSize];
-            long totalbytesRead = 0;
 
             SqlConnection sqlConnection = new SqlConnection(_connectionString);
             sqlConnection.Open();
             SqlTransaction sqlTransaction = null;
+
+            SqlCommand checkFileStreamEnabled = new SqlCommand("SELECT TOP 1 1 FROM sys.columns c WHERE OBJECT_SCHEMA_NAME(C.object_id) = 'LightDMS' AND OBJECT_NAME(C.object_id) = 'FileContent' AND c.Name = 'Content' AND c.is_filestream = 1", sqlConnection, sqlTransaction);
+            string createdDate = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fff");
+            bool fileStreamEnabled = checkFileStreamEnabled.ExecuteScalar() != null;
+            bool isS3 = _lightDmsOptions.UploadTarget == UploadTarget.S3;
+            bool isAzure = _lightDmsOptions.UploadTarget == UploadTarget.Azure;
             try
             {
                 sqlTransaction = sqlConnection.BeginTransaction(IsolationLevel.ReadUncommitted);
-
-                SqlCommand checkFileStreamEnabled = new SqlCommand("SELECT TOP 1 1 FROM sys.columns c WHERE OBJECT_SCHEMA_NAME(C.object_id) = 'LightDMS' AND OBJECT_NAME(C.object_id) = 'FileContent' AND c.Name = 'Content' AND c.is_filestream = 1", sqlConnection, sqlTransaction);
-                string createdDate = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fff");
-
-                if (checkFileStreamEnabled.ExecuteScalar() == null)
-                {   //FileStream is not supported
-                    SqlCommand createEmptyFileContent = new SqlCommand("INSERT INTO LightDMS.FileContent(ID, [Content], [CreatedDate]) VALUES('" + id + "', 0x0, '" + createdDate + "');", sqlConnection, sqlTransaction);
+                
+                if (!fileStreamEnabled ||

```

Comment:
Please remove `INSERT INTO LightDMS.FileContent` from Rhetos.LightDms.Storage.SqlFileStreamProvider, so that all upload variants insert FileContent here.